src/shared/Microsoft.Git.CredentialManager.Tests/StringExtensionsTests.cs

```diff
@@ -271,5 +271,28 @@ public void StringExtensions_TrimUntilLastIndexOf_String_Null_ThrowsArgumentNull
         {
             Assert.Throws<ArgumentNullException>(() => StringExtensions.TrimUntilLastIndexOf(null, "://"));
         }
+
+        [Theory]
+        [InlineData("fooTRIMbar", "TRIM", "foobar")]
+        [InlineData("fooTRIMbar", "trim", "fooTRIMbar")]
+        [InlineData("fooTRIMbar", "tRiM", "fooTRIMbar")]
+        public void StringExtensions_TrimMiddle_String(string input, string trim, string expected)
+        {
+            string actual = StringExtensions.TrimMiddle(input, trim);
```

Comment:
Does this just remove a particular string?  Does it have to be in the middle?  Wondering if RemoveString or similar would work better than TrimMiddle

Reply:
TrimMiddle because it doesn't trim from the end of the string, like the rest of the trim commands. Instead it 'cuts out' the matched text (i.e, it keeps the text before and after the matched string).